tsconfig.json

```diff
@@ -0,0 +1,12 @@
+{
+  "extends": "@vercel/style-guide/typescript",
+  "compilerOptions": {
+    "target": "esnext",
```

Comment:
target esnext requires quite new version of node and module esnext requires ESM. I feel we can keep the module as CJS output with es6 syntax, and support both ESM + CJS resolution in another PR

Reply:
What do you think about this?

https://github.com/vercel/serve/pull/706#discussion_r917546684

I don't think we can output CJS as some of the modules (e.g., `chalk`) we use are pure ESM.

Reply:
It brings quite a lot changes though IMO, you don't need to bump all the deps together. If we mark this repo as `module: "commonjs"` with no deps bumping, it can still be imported by ESM package

Reply:
The plan is to publish this as a breaking change.

Reply:
That sounds like a good idea.